spot-client/src/common/remote-control/P2PSignalingClient.js

```diff
@@ -0,0 +1,156 @@
+    }
+
+    /**
+     * Checks if the connection is ready to send commands.
+     *
+     * @returns {boolean}
+     */
+    isReady() {
+        const first = this._getFirstPeerConnection();
+
+        return first && first.isDataChannelActive();
+    }
+
+    /**
+     * Client processing for data channel messages. A client processes only acks.
+     *
+     * @param {string} remoteAddress - The remote address that sent the message.
+     * @param {Object} msg - JSON object received in message content.
+     * @private
+     * @returns {void}
+     */
+    _processDataChannelMessage(remoteAddress, msg) {
+        const { command, requestId } = msg;
+
+        if (command === 'ack' && requestId) {
+            const commandPromise = this._p2pSignalingRequests.get(requestId);
+
+            if (commandPromise) {
+                commandPromise.resolve();
+            } else {
+                logger.warn('No P2P command promise exists', { requestId });
+            }
+
+            this._p2pSignalingRequests.delete(requestId);
+        } else if (msg.command === 'status' && msg.newStatus) {
+            this.emit(P2PSignalingClient.SPOT_TV_STATUS_UPDATE, remoteAddress, msg.newStatus);
+        } else {
+            super._processDataChannelMessage(remoteAddress, msg);
+        }
+    }
+
+    /**
+     * Callback triggered by {@link PeerConnection} whenever data channel's readiness state changes.
+     *
+     * @param {string} remoteAddress - The remote address associated with source {@link PeerConnection}.
+     * @param {boolean} isReady - Is it ready or not.
+     * @protected
+     * @returns {void}
+     */
+    _onDCReadyStateChanged(remoteAddress, isReady) {
+        super._onDCReadyStateChanged(remoteAddress, isReady);
+
+        if (!this.isReady()) {
+
+            // NOTE: It might be possible to give it some time to reconnect
+            const count = this._p2pSignalingRequests.size;
+
+            count && logger.log('Rejecting signaling promises', { count });
+
+            for (const promise of this._p2pSignalingRequests.values()) {
+                promise.reject('P2P signaling channel closed');
+            }
+
+            this._p2pSignalingRequests.clear();
+        }
+    }
+
+    /**
+     * Sends remote control command over the underlying P2P data channel.
+     *
+     * @param {string} command - An RCS command to be sent.
+     * @param {string} data - Any data command specific.
+     * @returns {Promise<void>} - Resolved when an ack is received from the server or rejected if anything goes wrong.
+     */
+    sendCommand(command, data) {
+        logger.log('Sending command over P2P', {
+            command,
+            data
+        });
+
+        const peerConnection = this._getFirstPeerConnection();
+
+        if (!peerConnection) {
+            throw new Error('No PeerConnection');
+        }
+
+        // FIXME time may not be 100% reliable for unique IDs
+        const requestId = new Date().getTime();
```

Comment:
Why not use one of the other helpers in common for creating unique ids? Could it even be an increment or a combination of some id of the remote and an increment?

Reply:
Right, I will do that